### PR TITLE
mg master 1195 add OSPdO content incl 3 new modules

### PR DESCRIPTION
**Pull request by mickogeary**

### CODE REVIEW

File: doc-Service-Telemetry-Framework/modules/proc_deploying-the-overcloud-for-director-operator.adoc [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+. Login to your {OpenShift} environment and change to the project that hosts your {OpenStack} ({OpenStackShort}) deployment.
+
+. Open the `OpenStackConfigGenerator` custom resource for editing:
++
+[source,bash,options="nowrap",subs="verbatim"]
+----
+$ oc edit OpenStackConfigGenerator
+----
+
+. Add the `metrics/ceilometer-write-qdr.yaml` and `metrics/qdr-edge-only.yaml` environment files as values for the `heatEnvs` parameter, save your edits and close the `OpenStackConfigGenerator` custom resource.
```
Comment (rheslop) [inline doc-Service-Telemetry-Framework/modules/proc_deploying-the-overcloud-for-director-operator.adoc:19]:
```suggestion
. Add the `metrics/ceilometer-write-qdr.yaml` and `metrics/qdr-edge-only.yaml` environment files as values for the `heatEnvs` parameter. Save your edits, and close the `OpenStackConfigGenerator` custom resource:
```

I added a colon at the end as well, but the NOTE admonition gets in the way. Would it make sense to move it down past the example?

File: doc-Service-Telemetry-Framework/modules/proc_configuring-red-hat-openstack-platform-overcloud-for-stf-using-director-operator.adoc [hunk 2]
```diff
@@ -2,20 +2,26 @@
 = Deploying {OpenStack} overcloud for {Project} using director Operator
 
 [role="_abstract"]
-As part of the {OpenStack} ({OpenStackShort}) overcloud deployment using director Operator, you must configure the data collectors and the data transport to {Project} ({ProjectShort}).
+When you deploy the {OpenStack} ({OpenStackShort}) overcloud deployment using director Operator, you must configure the data collectors and the data transport to {Project} ({ProjectShort}).
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-red-hat-openstack-platform-overcloud-for-stf-using-director-operator.adoc:5]: Something about this seems off... 

> you must configure the data collectors and the data transport to {Project} ({ProjectShort}).

I think perhaps the "to {Project}" maybe should be "for {Project}"?

File: doc-Service-Telemetry-Framework/modules/proc_configuring-red-hat-openstack-platform-overcloud-for-stf-using-director-operator.adoc [hunk 3]
```diff
@@ -2,20 +2,26 @@
...
+
+.Prerequisites
+* You are familiar with deploying {OpenStackShort} with the {OpenStackShort} director Operator.
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-red-hat-openstack-platform-overcloud-for-stf-using-director-operator.adoc:8]:
```suggestion
* You are familiar with deploying and managing {OpenStackShort} with the {OpenStackShort} director Operator.
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-the-stf-connection-for-director-operator-for-the-overcloud.adoc [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+. Login to the {OpenShift} environment and change to the project that hosts your {OpenStack} deployment.
+
+. Open the `heat-env-config-deploy` ConfigMap for editing:
++
+[source,bash,options="nowrap",subs="verbatim"]
+----
+$ oc edit heat-env-config-deploy
+----
+
+. Add your `stf-connectors.yaml` configuration to the `heat-env-config-deploy ConfigMap`, appropriate to your environment, save your edits and close the file:
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-the-stf-connection-for-director-operator-for-the-overcloud.adoc:19]:
```suggestion
. Add your `stf-connectors.yaml` configuration to the `heat-env-config-deploy` ConfigMap, appropriate to your environment, save your edits and close the file:
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-the-stf-connection-for-director-operator-for-the-overcloud.adoc [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            cloud1-telemetry:
+                format: JSON
+                presettle: false
+
+        CollectdSensubilityResultsChannel: sensubility/cloud1-telemetry
+----
+
+[role="_additional-resources"]
+.Additional resources
+* For more information about the `stf-connectors.yaml` environment file, see link:https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/16.2/html/service_telemetry_framework_1.5/assembly-completing-the-stf-configuration_assembly#configuring-the-stf-connection-for-the-overcloud_assembly-completing-the-stf-configuration[Configuring the STF connection for the overcloud]
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-the-stf-connection-for-director-operator-for-the-overcloud.adoc:64]:
```suggestion
* For more information about the `stf-connectors.yaml` environment file, see xref:configuring-the-stf-connection-for-the-overcloud_assembly-completing-the-stf-configuration[].
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-the-stf-connection-for-director-operator-for-the-overcloud.adoc [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+* For more information about adding heat templates to a {OpenStack} director Operator deployment, see link:https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/16.2/html/rhosp_director_operator_for_openshift_container_platform/assembly_adding-heat-templates-and-environment-files-with-the-director-operator_rhosp-director-operator#doc-wrapper[Adding heat templates and environment files with the director Operator]
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-the-stf-connection-for-director-operator-for-the-overcloud.adoc:66]:
```suggestion
* For more information about adding heat templates to a {OpenStack} director Operator deployment, see link:{defaultURL}/rhosp_director_operator_for_openshift_container_platform/assembly_adding-heat-templates-and-environment-files-with-the-director-operator_rhosp-director-operator#doc-wrapper[Adding heat templates and environment files with the director Operator]
```

File: doc-Service-Telemetry-Framework/modules/proc_deploying-the-overcloud-for-director-operator.adoc [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  - <existing_environment_file_references>
+  - metrics/ceilometer-write-qdr.yaml
+  - metrics/qdr-edge-only.yaml
+----
+
+. If you already deployed a {OpenStack} environment using director Operator and generated a new `OpenStackConfigVersion`, edit the `OpenStackDeploy` object of your deployment, and set the value of `spec.configVersion` to the new `OpenStackConfigVersion` in order to update the overcloud deployment.
+
+[role="_additional-resources"]
+.Additional resources
+* For more information about getting the latest `OpenStackConfigVersion`, see link:https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/16.2/html/rhosp_director_operator_for_openshift_container_platform/assembly_configuring-overcloud-software-with-the-director-operator_rhosp-director-operator#proc_obtain-the-latest-openstackconfigversion.adoc_assembly_configuring-overcloud-software-with-the-director-operator[Obtain the latest OpenStackConfigVersion]
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_deploying-the-overcloud-for-director-operator.adoc:44]:
```suggestion
* For more information about getting the latest `OpenStackConfigVersion`, see link:{defaultURL}/rhosp_director_operator_for_openshift_container_platform/assembly_configuring-overcloud-software-with-the-director-operator_rhosp-director-operator#proc_obtain-the-latest-openstackconfigversion.adoc_assembly_configuring-overcloud-software-with-the-director-operator[Obtain the latest OpenStackConfigVersion]
```

File: doc-Service-Telemetry-Framework/modules/proc_deploying-the-overcloud-for-director-operator.adoc [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+* For more information about applying the overcloud configuration with director Operator, see link:https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/16.2/html/rhosp_director_operator_for_openshift_container_platform/assembly_configuring-overcloud-software-with-the-director-operator_rhosp-director-operator#proc_applying-overcloud-configuration-with-the-director-operator_assembly_configuring-overcloud-software-with-the-director-operator[Applying overcloud configuration with the director Operator]
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_deploying-the-overcloud-for-director-operator.adoc:46]:
```suggestion
* For more information about applying the overcloud configuration with director Operator, see link:{defaultURL}/rhosp_director_operator_for_openshift_container_platform/assembly_configuring-overcloud-software-with-the-director-operator_rhosp-director-operator#proc_applying-overcloud-configuration-with-the-director-operator_assembly_configuring-overcloud-software-with-the-director-operator[Applying overcloud configuration with the director Operator]
```

**Review by leifmadsen — CHANGES_REQUESTED** [review]
Added some comments and updates to links to not have them hard-coded to the RHOSP versions. I also am thinking maybe wrapping the top level includes for this so it's generated only downstream. I'm not sure how readily available OSPdO is even upstream, and it's not important to show this in the upstream documentation for STF, in my opinion.

File: doc-Service-Telemetry-Framework/modules/proc_configuring-red-hat-openstack-platform-overcloud-for-stf-using-director-operator.adoc (hunk 2 above)
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-red-hat-openstack-platform-overcloud-for-stf-using-director-operator.adoc:5]:
```suggestion
When you deploy the {OpenStack} ({OpenStackShort}) overcloud deployment using director Operator, you must configure the data collectors and the data transport for {Project} ({ProjectShort}).
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-red-hat-openstack-platform-overcloud-for-stf-using-director-operator.adoc [hunk 9]
```diff
@@ -2,20 +2,26 @@
...
 //endif::include_when_13,include_when_17[]
-. xref:retrieving-the-qdr-route-address_assembly-completing-the-stf-configuration-using-director-operator[Retrieving the {MessageBus} route address]
-//. xref:creating-the-base-configuration-for-stf_assembly-completing-the-stf-configuration[Creating the base configuration for {ProjectShort}]
-//. xref:configuring-the-stf-connection-for-the-overcloud_assembly-completing-the-stf-configuration[Configuring the {ProjectShort} connection for the overcloud]
-//. xref:deploying-the-overcloud_assembly-completing-the-stf-configuration[Deploying the overcloud]
+
+. xref:retrieving-the-qdr-route-address_assembly-completing-the-stf-configuration[Retrieving the {MessageBus} route address]
+. xref:proc_creating-the-base-configuration-for-director-operator-for-stf[Creating the base configuration for director Operator for {ProjectShort}]
+. xref:proc_configuring-the-stf-connection-for-director-operator-for-the-overcloud[Configuring the {ProjectShort} connection for the overcloud]
+. xref:dproc_deploying-the-overcloud-for-director-operator[Deploying the overcloud for director operator]
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-red-hat-openstack-platform-overcloud-for-stf-using-director-operator.adoc:20]:
```suggestion
. xref:proc_deploying-the-overcloud-for-director-operator[Deploying the overcloud for director operator]
```

**Review by leifmadsen — APPROVED** [review]
Worked this with Michael today, and we're happy to land this.

**Pull request merged**